api.go

```diff
@@ -70,8 +63,12 @@ func WithSettingEngine(s SettingEngine) func(a *API) {
 
 // WithInterceptorRegistry allows providing Interceptors to the API.
 // Settings should not be changed after passing the registry to an API.
-func WithInterceptorRegistry(interceptorRegistry *interceptor.Registry) func(a *API) {
+func WithInterceptorRegistry(ir *interceptor.Registry) func(a *API) {
```

Comment:
Can you add a test to confirm the new behavior? `WithSettingRegistry` that is nil shouldn't segfault